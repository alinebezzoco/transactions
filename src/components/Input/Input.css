input {
  height: 1.875rem;
  width: auto;
  border-radius: 3.125rem;
  font-size: 0.875rem;
  border-style: none;
  border: 1px solid var(--color-purple);
  padding: 0 0.625rem;
  margin-bottom: 1.25rem;
}

@media (min-width: 768px) {
  input {
    margin-bottom: 0;
  }
}

label {
  padding: 0 0.625rem 0 1.25rem;
  font-weight: 700;
  text-align: left;
}
