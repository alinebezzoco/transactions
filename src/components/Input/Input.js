import React from "react";
import PropTypes from 'prop-types';
import "./Input.css";

const Input = props => {
  return (
    <div className="form-group">
      <label htmlFor={props.name} className="form-label">
        {props.title}
      </label>
      <input
        className="form-input"
        id={props.name}
        name={props.name}
        type={props.type}
        value={props.value}
        onChange={props.handleChange}
        placeholder={props.placeholder}
        required={props.required}
      />
    </div>
  );
};

Input.propTypes = {
  name: PropTypes.string,
  type: PropTypes.string, 
  value: PropTypes.any, 
  handleChange: PropTypes.func, 
  placeholder: PropTypes.string, 
  required: PropTypes.bool
};

export default Input;
