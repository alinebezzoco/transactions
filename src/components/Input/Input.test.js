import React from "react"
import Enzyme, { shallow } from "enzyme"
import Adapter from "enzyme-adapter-react-16"
import Input from "./Input"

describe("Input Component", () => {
	beforeAll(() => {
		Enzyme.configure({ adapter: new Adapter() })
	})

	it("renders correctly", () => {
		const component = shallow(<Input id={1} title={"Description"} name={"description"} type={"text"} value={"Hamburgueria"} placeholder={"Digite um nome"} />)
		expect(component).toMatchSnapshot()
	})
})