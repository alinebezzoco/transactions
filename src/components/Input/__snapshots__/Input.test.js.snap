// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component renders correctly 1`] = `ShallowWrapper {}`;
